server.js

```diff
@@ -14,10 +16,92 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+const messages = [];
+let nextMessageID = 0;
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.post("/message", (req, res) => {
+  if (req.body.from !== "" && req.body.text !== "") {
+    nextMessageID++;
+    const message = {};
+    message.id = nextMessageID;
+    message.from = req.body.from;
+    message.text = req.body.text;
+    // message.timeSent = Date.now();
+    // message.timeSent = new Date().toDateString();
+    // message.timeSent = new Date().toISOString();
+    message.timeSent = new Date().toString();
+    // message.timeSent = new Date().toLocaleDateString();
+    messages.push(message);
+    res.send(messages);
+  } else {
+    res.sendStatus(400);
+  }
+});
+
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.get("/message/:id", (req, res) => {
+  const userId = req.params.id;
```

Comment:
The ID is the message ID not the user ID so easier to understand if it's named that way.